community/index/src/main/java/org/neo4j/index/gbptree/StructurePropagation.java

```diff
@@ -25,9 +25,16 @@
  *
  * @param <KEY> type of key.
  */
-class SplitResult<KEY>
+class StructurePropagation<KEY>
```

Comment:
i would propose to rename it to SplitContext or UpdateContext, and also include in it generational information for example to simplify InternalTreeLogic methods signatures, basically use it as context object instead.